File: Week_03/G20200343040030/binary-tree-preorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var preorderTraversal = function(root) {
  let res = [];
  function helper(root) {
    if (!root) return;
    res.push(root.val);
    helper(root.left);
    helper(root.right);
  }

  helper(root);

  return res;
};

File: Week_03/G20200343040156/BinaryTreePreorderTraversal.java
class BinaryTreePreorderTraversal {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> re = new ArrayList<>();
        helper(re,root);
        return re;
    }
    public List<Integer> helper(List<Integer> re ,TreeNode root){

        if(null!=root){
            re.add(root.val);
            if(null!=root.left){
                helper(re,root.left);
            }
            if(null!=root.right){
                helper(re,root.right);
            }
        }
        return re;
    }
}